go/enclave/events/subscription_manager.go

```diff
@@ -169,26 +169,27 @@ func (s *SubscriptionManager) encryptLogs(logsByID map[gethrpc.ID][]*types.Log)
 	return encryptedLogsByID, nil
 }
 
-// Extracts the (potential) user addresses from the topics. If there is no code associated with an address, it's a user
-// address.
+// Extracts the user addresses from the topics.
 func getUserAddrsFromLogTopics(log *types.Log, db *state.StateDB) []string {
-	var userAddrs []string //nolint:prealloc
+	var userAddrs []string
 
 	for _, topic := range log.Topics {
-		// Since addresses are 20 bytes long, while hashes are 32, only topics with 12 leading zero bytes can
-		// (potentially) be user addresses.
-		topicHex := topic.Hex()
-		if topicHex[2:len(zeroBytesHex)+2] != zeroBytesHex {
-			continue
+		potentialAddr := gethcommon.HexToAddress(topic.Hex())
+
+		// An address is considered a user address if all the following are true:
```

Comment:
umm, still a bit confusing. sorry.

1. start with potential address
2. check format (leading 0). Fail -> exit
3. check whether it exists in the db
 db.exist || db.balance!=0 || db.nonce !=0. 
4. If yes . Then check if the db.code is nil 
5. if yes, then append

I think it's better to express this way as it highlights the related checks